packages/@aws-cdk/aws-certificatemanager/lib/dns-validated-certificate.ts

```diff
@@ -160,3 +172,11 @@ export class DnsValidatedCertificate extends CertificateBase implements ICertifi
     return errors;
   }
 }
+
+// https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/specifying-rrset-conditions.html
+function addWildcard(domainName: string) {
+  if (domainName.startsWith('*.')) {
+    return domainName;
+  }
+  return `*.${domainName}`;
```

Comment:
In the [examples](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/specifying-rrset-conditions.html#route53_rrset_conditionkeys-examples) it uses `\\052` instead of `*`. The way I am reading it (I could be wrong), `*` is used to match part of the name (i.e. `*-beta.example.com`), but to match a domain name containing a wildcard you have to use `\\052`

```json
"Condition": {
                "ForAllValues:StringEquals":{
                     "route53:ChangeResourceRecordSetsNormalizedRecordNames": ["\\052.example.com"]
                }
            }
```

Reply:
I'm using a `StringLike` condition here (not a `StringEquals`) with the IAM wildcard operator (not the wildcard of a domain- because the ACM validation record is <something>.<domain name>, e.g. if you want to validate `cool.domain.com`, the record will be of the form `_b275f33adafe0d301e18c9d320a99971.cool.domain.com` and if you want a certificate for `*.domain.com` it will be `_b275f33adafe0d301e18c9d320a99971.domain.com`